accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,395 @@
+
+E.g. The output is presented in the "CaaS Platform > cluster name" section:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+cp0       SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+#### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (drained and cordoned off): [using `kubectl`](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#use-kubectl-drain-to-remove-a-node-from-service) from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt. NOTE: this removal is permanent and the cluster cannot join any cluster in the future. A complete reinstallation is required and Uyuni/SUSE Manager must offer the user to re-provision the node from scratch, e.g. via AutoYaST.
+
+Example state:
+
+```
+remove_node:
+      cmd.run:
+        - name: skuba node remove <nodename>
+```
+
+NOTE: Other `skuba` options can be implemented in the future (e.g. resetting and reconfiguring a node).
+
+#### Cluster upgrade
+
+All the cluster updates must be handled by `skuba`.
+
+Uyuni/SUSE Manager can check and notify the user if an update for the cluster is present by triggering `skuba cluster upgrade plan` on the Management Node (using Salt) and check if a new output is available in a cron fashion (e.g. Taskomatic job).
+
+Uyuni/SUSE Manager will then interface with the Management node and upgrade every control plane in the cluster first:
+
+`skuba node upgrade apply --target <control-plane-node-ip> --user <user> --sudo`
+
+and then every worker:
+
+`skuba node upgrade apply --target <worker-node-ip> --user <user> --sudo`
+
+This can be done by supplying the command to the Management Node via Salt.
+
+NOTE: CaaS Platform stack is also deployed via containers. We will need to update this section when the CaaS Platform team concludes the "add-ons update story".
+
+#### CVE Auditing
+
+By leveraging the existing feature of CVE Auditing, Uyuni/SUSE Manager can detect and notify the user whether the cluster is running a vulnerable image. When Uyuni/SUSE Manager rebuilds the image and patches it, Uyuni/SUSE Manager can also guide the user in modifying the Kubernetes deployment to pull the updated image (in case of `imagePullPolicy:` is not `Always`) or do a `kubectl rolling-update`.
+
+##  Future work
+
+### CVE Auditing
+
+A useful addition would be to have Uyuni/SUSE Manager as an admission controller for rejecting pod creation whenever an image used is vulnerable.
+
+### Deploy OpenStack Cloud
+
+If OpenStack Cloud will be deployed using AirShip (still under discussion), the approach described in this RFC can be further reused for deploying and managing an OpenStack Cloud, given that `skuba` is similar to `airshipctl`:
```

Comment:
> My concern is that SUMA wants to have a deep understanding of applications. This means a knowledge of all those applications are wired together. This can become quite complex. There is already a tool that does that: airship. In that case, we should take airship as a whole ecosystem for app delivery from bare metal to the app.
> However, if we are considering that suma only has a knowledge of "clusters" and does not deal with what's running on top (leaving that to each product), then it simplifies the view, and allow more flexibility for customers.

I think the direction we are going with this RFC is exactly the latter: SUMA has just knowledge of deploying and managing clusters and does not know what is running on top.

Reply:
Ok that's re-assuring. Some of the text above suggests otherwise (not this sentence though), which I think would deserve rephrasing.

Reply:
Added a clarification note.